### DISCUSSION TITLE
UI For Seeing all Objects in a Workflow State - Unpublished Changes

### DISCUSSION THREAD

**Discussion by cancan101** [pimcore/pimcore#16945] [Q&A / Support]
What is the best way to list all of the Objects in a given Workflow State? of note, I these Objects may have unpublished changes as I am currently only publishing at the end of the workflow?

In this case, we are _saving_ between workflow steps and publishing (saving to DB) only at the _end_. I would still like to provide my users some UI where they can see all the Objects at a given state in the workflow.

I don't think the standard grid view with a filter will work due to https://github.com/pimcore/pimcore/issues/5322.

**Comment by markus-moser** [comment]
This will not be that easy and is dependent on your concrete implementation of the workflow but you could use a custom SQL report for this. When you use the default `state_table` marking store this should work without a problem. When you store the state information in data object fields you will have the problem that unpublished state changes then are only stored in the version (and therefore in the filesystem) and not in the database. In this case you would need to hook into the workflow engine events and listen to all state changes and store them into a custom additional table. Without a additional data store it would be impossible to filter it in that case.

**Reply by cancan101** [reply]
These are Objects so currently we are using `single_state`. I posted a related question #16962 which might be a bit of a hack, but wondering if there is any viability there?

**Reply by markus-moser** [reply]
Yes then you would need to either save the state additionally in a separate table (based on events) or do somehow the same approach like in the other task to store the state already in the database but do not publish the version.
